modules/shredder/src/main/scala/com.snowplowanalytics.snowplow.storage/spark/EventUtils.scala

```diff
@@ -100,12 +100,13 @@ object EventUtils {
       transformBool,
       _ => json.show,
       escape,
-      _ => json.noSpaces,
-      _ => json.noSpaces)
+      _ => escape(json.noSpaces),
+      _ => escape(json.noSpaces))
 
   /** Prevents data with newlines and tabs from breaking the loading process */
   private def escape(s: String): String =
-    if (s == NullCharacter) "\\\\N" else s
+    if (s == NullCharacter) "\\\\N"
+    else s.replace('\t', ' ').replace('\n', ' ')
```

Comment:
```suggestion
    else s.replace('\t', '\\t').replace('\n', '\\n')
```
To not lose info and be able to reconstruct the original event?

Reply:
That's actually quite interesting idea.

I just recovered the pre-R33 behavior that nobody complained about, but you made me doubt it was the best one from the beginning. I think values with tabs and newlines appear in enriched data more by mistake and as exception rather than a normal situation and rarely used for actual analysis (let alone text normalization), but it still doesn't alllow us to corrupt it.

I think I'll leave it as is in R34, but willl create a ticket to consider to switching to this behavior.